src/ray/gcs/gcs_client/service_based_gcs_client.h

```diff
@@ -70,7 +70,7 @@ class RAY_EXPORT ServiceBasedGcsClient : public GcsClient {
   // A timer used to check if gcs server address changed.
   std::unique_ptr<boost::asio::deadline_timer> detect_timer_;
   std::function<bool(std::pair<std::string, int> *)> get_server_address_func_;
-  std::function<void(bool)> resubscribe_func_;
+  std::function<void(bool)> restart_functions_;
```

Comment:
recovery_functions_?

Reply:
`recovery_function` sounds like the one who holds gcs client recovered from a failure. Maybe `reconnect_functions` is better.

Avoid being involved into naming discussions, I just pull the resend logic from the functions as there's only one.